packages/cursorless-engine/src/util/uniqWithHash.ts

```diff
@@ -0,0 +1,74 @@
+import { uniqWith, isEqual } from "lodash";
+
+/**
+ * Like lodash.uniqWith, but uses a hash function to (mostly) avoid quadratic runtime.
+ * @param array The array to uniq
+ * @param fn The equality function
+ * @param hash The hash function. It must be a valid hash function, insofar as it must return the same value for equal items. A hash function that returns a constant value is equivalent to lodash.uniqWith.
+ * @returns The uniq array
+ */
+export function uniqWithHash<T>(
+  array: T[],
+  fn: (a: T, b: T) => boolean,
```

Comment:
I might rename this `isEqual` just to make it super clear what it does?

Reply:
done